File: mindvault-ai-python/main/python/models/rerank_dto.py
from typing import List, Optional
from pydantic import BaseModel, Field
from main.python.models.rerank_source_item import RerankSourceItem

class RerankDTO(BaseModel):
    query: str # 用户搜的关键词
    sources: List[RerankSourceItem] # 待排序的文档列表
    top_k: Optional[int] = Field(default=5, alias="topK") # 返回前几条
    user_id: Optional[str] = Field(default=None, alias="userId") # 谁在搜（可选）

File: mindvault-ai-api/src/main/java/org/example/mindvaultaiapi/dto/RerankDTO.java
package org.example.mindvaultaiapi.dto;


import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class RerankDTO {

    private String query;// 用户搜索的关键词
    private List<RerankSourceItem> sources;// 待排序的文档列表
    private int topK; // 返回的前几条排序结果数量
    private String userId; // 谁在搜索
}
